File: twosum2/main.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        hashmap = {}
        for i, num in enumerate(numbers):
            diff = target - num
            if diff in hashmap:
                return [hashmap[diff], i+1]
            
            hashmap[num] = i + 1
        
        return []

File: twosum2/main.go
package twosum2

// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

func twoSum(numbers []int, target int) []int {
	hashMap := make(map[int]int)
	for i, num := range numbers {
		diff := target - num
		if _, ok := hashMap[diff]; ok {
			return []int{hashMap[diff], i+1}
		}

		hashMap[num] = i + 1
	}

	return []int{}
}
